diff --git a/boost/type_traits/is_unsigned.hpp b/boost/type_traits/is_unsigned.hpp
--- a/boost/type_traits/is_unsigned.hpp
+++ b/boost/type_traits/is_unsigned.hpp
@@ -14,6 +14,7 @@
 #include <boost/type_traits/ice.hpp>
 #include <boost/type_traits/remove_cv.hpp>
 #include <boost/type_traits/conditional.hpp>
 
 #include <climits>
+#include <type_traits>
 
 namespace boost {
 
@@ -35,8 +36,13 @@
    // the correct answer.
    //
    typedef typename remove_cv<T>::type no_cv_t;
-   static const no_cv_t minus_one = (static_cast<no_cv_t>(-1));
-   static const no_cv_t zero = (static_cast<no_cv_t>(0));
+   typedef typename boost::conditional<
+      boost::is_enum<no_cv_t>::value,
+      typename std::underlying_type<no_cv_t>::type,
+      no_cv_t
+   >::type test_t;
+
+   static const test_t minus_one = static_cast<test_t>(-1);
+   static const test_t zero      = static_cast<test_t>(0);
 };
 
 template <class T>
