diff --git a/src/common/command_line.h b/src/common/command_line.h
index d80a1b0df..d58de89d2 100644
--- a/src/common/command_line.h
+++ b/src/common/command_line.h
@@ -184,7 +184,8 @@ namespace command_line
   }
 
   template<typename T, bool required, bool dependent, int NUM_DEPS>
-  void add_arg(boost::program_options::options_description& description, const arg_descriptor<T, required, dependent, NUM_DEPS>& arg, bool unique = true)
+  typename std::enable_if<!std::is_same<T, bool>::value, void>::type
+  add_arg(boost::program_options::options_description& description, const arg_descriptor<T, required, dependent, NUM_DEPS>& arg, bool unique = true)
   {
     if (0 != description.find_nothrow(arg.name, false))
     {
@@ -207,8 +208,7 @@ namespace command_line
     description.add_options()(arg.name, make_semantic(arg, def), arg.description);
   }
 
-  template<>
-  inline void add_arg(boost::program_options::options_description& description, const arg_descriptor<bool, false>& arg, bool unique)
+  inline void add_arg(boost::program_options::options_description& description, const arg_descriptor<bool, false>& arg, bool unique = true)
   {
     if (0 != description.find_nothrow(arg.name, false))
     {
